src/main/java/fakebot/FakeBot.java

```diff
@@ -0,0 +1,313 @@
+
+        return "";
+    }
+
+    /**
+     * Process Done Command.
+     *
+     * @param command Command to Process.
+     * @return Returns process Done command result message
+     */
+    private String processDoneCommand(Command command) {
+        int doneIndex = Integer.parseInt(command.getDescription()) - 1;
+        taskList.getTask(doneIndex).markComplete();
+        saveHistory();
+        return getDoneMessage(taskList.getTask(doneIndex));
+    }
+
+    /**
+     * Process Todos Command.
+     *
+     * @param command Command to Process.
+     * @return Returns process ToDos command result message
+     */
+    private String processTodoCommand(Command command) {
+        ToDos todoTask = new ToDos(command.getDescription());
+        taskList.addTask(todoTask);
+        saveHistory();
+        return getAddedTaskMessage(todoTask);
+    }
+
+    /**
+     * Process Deadline Command.
+     *
+     * @param command Command to Process.
+     * @return Returns process Deadline command result message
+     */
+    private String processDeadlineCommand(Command command) {
+        String[] deadlineDetails = command.getDescription().split(DEADLINE_SPLIT_REGEX);
+        String[] dates = deadlineDetails[1].split(" ");
+        LocalDate date = LocalDate.parse(dates[0]);
+        LocalTime time = LocalTime.parse(dates[1]);
+        Deadlines deadlineTask = new Deadlines(deadlineDetails[0], date, time);
+        taskList.addTask(deadlineTask);
+        saveHistory();
+        return getAddedTaskMessage(deadlineTask);
+    }
+
+    /**
+     * Process Event Command.
+     *
+     * @param command Command to Process.
+     * @return Returns process Event command result message
+     */
+    private String processEventCommand(Command command) {
+        String[] eventDetails = command.getDescription().split(EVENT_SPLIT_REGEX);
+        String[] eventDates = eventDetails[1].split(" ");
+        LocalDate startDate = LocalDate.parse(eventDates[0]);
+        LocalTime startTime = LocalTime.parse(eventDates[1]);
+        LocalDate endDate = LocalDate.parse(eventDates[2]);
+        LocalTime endTime = LocalTime.parse(eventDates[3]);
+        Events eventTask = new Events(eventDetails[0], startDate, startTime, endDate, endTime);
+        taskList.addTask(eventTask);
+        saveHistory();
+        return getAddedTaskMessage(eventTask);
+    }
+
+    /**
+     * Process Delete Command.
+     *
+     * @param command Command to Process.
+     */
+    private String processDeleteCommand(Command command) {
+        int deleteIndex = Integer.parseInt(command.getDescription()) - 1;
+        Task deletedTask = taskList.getTask(deleteIndex);
+        taskList.removeTask(deleteIndex);
+        saveHistory();
+        return getDeleteMessage(deletedTask);
+    }
+
+    /**
+     * Validates User Input.
+     *
+     * @param command Command String that is yet to be parsed.
+     */
+    public Command validateCommand(String command) throws CommandException {
```

Comment:
Perhaps you could have used smaller helper methods to complete this method? Since there are big chunks of code that can be difficult to follow.

Reply:
will fix. thank you.